### PR TITLE
auth on connect and always call acknowledge response function

### PR DESCRIPTION
**Pull request by bucko13**
These are changes necessary for the socketManager to work on bPanel. `auth` will now always be `call`ed on connect. One possible downside of this is that it fails if the socket being connected to doesn't have an auth call. 

This also adds support for the namespace option when making connections. Allows the socket to indicate which namespace (i.e. client in the case of bPanel) it wants to connect to.

Couple thoughts: 
1) the auth on `handleSocket` is the convention that's always used in bcoin so maybe it's good to enforce it. 
2) We could only do auth if there is an apiKey in the payload. in bcoin world though even with `noAuth` and no apiKey you still have to call auth to initiate the socket listeners. I'm torn between it being better to expect there to be some auth and the fact that the client is expecting some additional setup that is handled by the server (this utility can/should be somewhat generic, i.e. not just for bpanel).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -57,3 +57,8 @@ typings/
 # dotenv environment variables file
 .env
 
+# system files
+.DS_Store
+
+# project files
+index.js
```
Comment (tynes) [inline .gitignore:64]: Why `index.js`?

Comment (tynes) [reply .gitignore:64]: Is it the output of a build step?

File: bsockMiddleware.js [hunk 2]
```diff
@@ -24,8 +24,8 @@ export default function bsockMiddleware (options) {
           socket.close();
         }
         // if it has a `connect` action then we connect to the bcoin socket
-        const { port, host, ssl, protocols } = action.bsock;
-        socket = bsock.connect(port, host, ssl, protocols);
+        const { port, host, ssl, protocols, namespace, apiKey } = action.bsock;
```
Comment (tynes) [inline bsockMiddleware.js:27]: Will the api key will show up in redux devtools if done this way?

Comment (tynes) [reply bsockMiddleware.js:27]: I suppose that isn't the worst, as the devtools shouldn't be built into the app everywhere, but other plugins could scrape any actions

File: bsockMiddleware.js [hunk 3]
```diff
@@ -35,7 +35,8 @@ export default function bsockMiddleware (options) {
           dispatch({ type: 'SOCKET_ERROR', payload: err });
         });
 
-        socket.on('connect', () => {
+        socket.on('connect', async () => {
+          await socket.call('auth', apiKey);
```
Comment (tynes) [inline bsockMiddleware.js:39]: Hopefully the socket was initialized with `ssl`

File: .gitignore (hunk 1 above)
Comment (tynes) [reply .gitignore:64]: I see here: https://github.com/bucko13/bsock-middleware/blob/master/package.json#L8

File: bsockMiddleware.js (hunk 3 above)
Comment (bucko13) [reply bsockMiddleware.js:39]: 😬

**Pull request merged**